### ISSUE TITLE
#cmakedefine01 macros are not documented

### ISSUE THREAD

**Issue by vardarirrgan1** [doxygen/doxygen#10498]
https://github.com/doxygen/doxygen/issues/3519 addressed macros generated by CMake using `#cmakedefine`. CMake also has a related feature called `#cmakedefine01`. Ref https://cmake.org/cmake/help/latest/command/configure_file.html

Could we get `#cmakedefine01` to be treated the same as `#cmakedefine`?

**Version**
1.9.8

**Comment by albert-github** [comment]
For testing purposes:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem (with the `#cmakdefifine01` and prefereably also with `@cmakedefine`)? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by albert-github** [comment]
I tried to create a small example and came up with:
**CMakeLists.txt**
```
project(cmakedefine)
cmake_minimum_required(VERSION 3.22)

set(HAVE_GETOPT "1")
set(ZERO_HAVE_GETOPT "0")

configure_file(${CMAKE_SOURCE_DIR}/aa.h.in ${CMAKE_SOURCE_DIR}/aa.h)
```
**aa.h.in**
```
/// \file
/// some tests for `#cmakedefine`

#cmakedefine01 HAVE_GETOPT
#cmakedefine01 ZERO_HAVE_GETOPT
#cmakedefine01 NOT_HAVE_GETOPT
```
running cmake wil give:
**aa.h**
```
/// \file
/// some tests for `#cmakedefine`

#define HAVE_GETOPT 1
#define ZERO_HAVE_GETOPT 0
#define NOT_HAVE_GETOPT 0
```

so the best doxygen can do would be:
```
/// \file
/// some tests for `#cmakedefine`

#define HAVE_GETOPT 0
#define ZERO_HAVE_GETOPT 0
#define NOT_HAVE_GETOPT 0
```
(note the difference for `HAVE_GETOPT`)

**Comment by vardarirrgan1** [comment]
I should have provided a more complete description. As in issue #3519, the use case is to document a file before substitution instead of after. When documenting preprocesser definitions, Doxygen currently treats `#cmakedefine` as if it were `#define`. My request is to do the same with `#cmakedefine01`

### Example

aa.in.h:
```
/// \file
/// some tests for `#cmakedefine`

/// A CMake configured macro
#cmakedefine HAVE_GETOPT

/// Another CMake configured macro
#cmakedefine01 ZERO_HAVE_GETOPT

/// A missing CMake variable. Should generate docs even if NOT_HAVE_GETOPT is not a CMake variable because of the behavior of #cmakedefine01.
#cmakedefine01 NOT_HAVE_GETOPT

/// Another missing CMake variable. Generates Doxygen output when run against the input file, but not when run against the output file because substitution does not produce a `#define`.
#cmakedefine OTHER_NOT_HAVE_GETOPT
```

We would like this to result in documentation for the first three cases when processing `aa.in.h`, so that Doxygen can be run against the source tree without having to first run CMake to configure the generated header.

**Comment by albert-github** [comment]
've just pushed a proposed patch, pull request #10502

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
